crates/re_ui/src/list_item2/property_content.rs

```diff
@@ -0,0 +1,278 @@
+
+struct PropertyActionButton<'a> {
+    icon: &'static crate::icons::Icon,
+    on_click: Box<dyn FnOnce() + 'a>,
+}
+
+/// [`ListItemContent`] to display property-like, two-column content, with the left column
+/// containing a label (along with an optional icon) and the right column containing some custom
+/// value (which may be editable).
+pub struct PropertyContent<'a> {
+    label: egui::WidgetText,
+    icon_fn: Option<Box<IconFn<'a>>>,
+    summary_only: bool,
+    value_fn: Option<Box<PropertyValueFn<'a>>>,
+    //TODO(ab): in the future, that should be a `Vec`, with some auto expanding mini-toolbar
+    action_buttons: Option<PropertyActionButton<'a>>,
+    /**/
+    //TODO(ab): icon styling? link icon right of label? clickable label?
+}
+
+impl<'a> PropertyContent<'a> {
+    pub fn new(label: impl Into<egui::WidgetText>) -> Self {
+        Self {
+            label: label.into(),
+            icon_fn: None,
+            summary_only: true,
+            value_fn: None,
+            action_buttons: None,
+        }
+    }
+
+    /// Provide an [`Icon`] to be displayed on the left of the label.
+    #[inline]
+    pub fn with_icon(self, icon: &'a Icon) -> Self {
+        self.with_icon_fn(|_, ui, rect, visuals| {
+            let tint = visuals.fg_stroke.color;
+            icon.as_image().tint(tint).paint_at(ui, rect);
+        })
+    }
+
+    /// Provide a custom closure to draw an icon on the left of the item.
+    #[inline]
+    pub fn with_icon_fn<F>(mut self, icon_fn: F) -> Self
+    where
+        F: FnOnce(&ReUi, &mut egui::Ui, egui::Rect, egui::style::WidgetVisuals) + 'a,
+    {
+        self.icon_fn = Some(Box::new(icon_fn));
+        self
+    }
+
+    /// Right aligned action button.
+    ///
+    /// Note: for aesthetics, space is always reserved for the action button.
+    // TODO(ab): accept multiple calls for this function for multiple actions. In that case, a `…´
+    // button should be displayed that turns into a mini-popup with all available actions
+    // TODO(ab): if ALL item in a scope have no button active, then we could skip reserving the
+    // space in the right margin.
+    #[inline]
+    pub fn action_button(
+        mut self,
+        icon: &'static crate::icons::Icon,
+        on_click: impl FnOnce() + 'a,
+    ) -> Self {
+        self.action_buttons = Some(PropertyActionButton {
+            icon,
+            on_click: Box::new(on_click),
+        });
+        self
+    }
+
+    /// Display value only for leaf or collapsed items.
+    ///
+    /// When enabled, the value for this item is not displayed for uncollapsed hierarchical items.
+    /// This is convenient when the value serves are a summary of the child content, which doesn't
+    /// need to be displayed when said content is visible.
+    ///
+    /// Enabled by default.
+    #[inline]
+    pub fn summary_only(mut self, summary_only: bool) -> Self {
+        self.summary_only = summary_only;
```

Comment:
Maybe `show_only_when_collapsed` is a better name?

Reply:
agreed